### PR TITLE
Add extended header feature for additional information on excel

### PR DESCRIPTION
**Pull request by JustinWonjaePark**
## Description
1. Add parameter for excel and csv writing
2. Differentiate exact matching and starting with sheet names

## Type of change
Please insert 'x' one of the type of change.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/fosslight_util/write_excel.py [hunk 1]
```diff
@@ -82,18 +82,29 @@ def remove_empty_sheet(sheet_items):
     return success, final_sheet_to_print
 
 
-def get_header_row(sheet_name, sheet_content):
+def get_header_row(sheet_name, sheet_content, extended_header):
```
Comment (soimkim) [inline src/fosslight_util/write_excel.py:85]: Let's change` extended_header `to `extended_header={}`.

File: src/fosslight_util/write_excel.py [hunk 2]
```diff
@@ -82,18 +82,29 @@ def remove_empty_sheet(sheet_items):
...
     selected_header = []
+
+    if extended_header:
```
Comment (soimkim) [inline src/fosslight_util/write_excel.py:88]: Even if `extended_header ` is` {}`, `if extended_header `will be True.
Let's remove `if extended_header `.

**Review by soimkim — CHANGES_REQUESTED** [review]

File: src/fosslight_util/write_excel.py [hunk 3]
```diff
@@ -82,18 +82,29 @@ def remove_empty_sheet(sheet_items):
...
+        for header_key in extended_header.keys():
+            _HEADER[header_key] = extended_header[header_key]
+
     for header_key in _HEADER.keys():
```
Comment (soimkim) [inline src/fosslight_util/write_excel.py:92]: I suggest the following.
- AS-IS: 
```
for header_key in _HEADER.keys():
        if header_key in sheet_name:
        if header_key == sheet_name:
            selected_header = _HEADER[header_key]
            break
```
- TO-BE:
```
selected_header = _HEADER[sheet_name]
```

File: src/fosslight_util/write_excel.py [hunk 4]
```diff
@@ -82,18 +82,29 @@ def remove_empty_sheet(sheet_items):
     return success, final_sheet_to_print
 
 
-def get_header_row(sheet_name, sheet_content):
+def get_header_row(sheet_name, sheet_content, extended_header):
     selected_header = []
+
```
Comment (soimkim) [inline src/fosslight_util/write_excel.py:87]: I suggest the following. 
- AS-IS :
```
    if extended_header:
        for header_key in extended_header.keys():
            _HEADER[header_key] = extended_header[header_key]
```
- TO-BE: https://pypi.org/project/jsonmerge
```
from  jsonmerge import merge
result = merge(_HEADER, extended_header)
```

Comment (soimkim) [reply src/fosslight_util/write_excel.py:87]: For TO-BE, `from jsonmerge import merge` should be added at the top of this file.
And `jsonmerge` should be added to requirements.txt.

File: src/fosslight_util/write_excel.py (hunk 1 above)
Comment (JustinWonjaePark) [reply src/fosslight_util/write_excel.py:85]: Changed as suggested.

**Comment by JustinWonjaePark** [comment]
Replace the PR with #43

File: src/fosslight_util/write_excel.py (hunk 4 above)
Comment (JustinWonjaePark) [reply src/fosslight_util/write_excel.py:87]: Changed as suggested.

File: src/fosslight_util/write_excel.py (hunk 3 above)
Comment (JustinWonjaePark) [reply src/fosslight_util/write_excel.py:92]: Removed if statement and change how to assign selected_header as suggested.

**Pull request closed without merging**